auditbeat/module/auditd/audit_linux.go

```diff
@@ -140,14 +156,32 @@ func (ms *MetricSet) Run(reporter mb.PushReporterV2) {
 		}()
 	}
 
-	for {
-		select {
-		case <-reporter.Done():
-			return
-		case msgs := <-out:
-			reporter.Event(buildMetricbeatEvent(msgs, ms.config))
+	// Spawn the stream buffer consumers
+	numConsumers := ms.config.StreamBufferConsumers
+	// By default (stream_buffer_consumers=0) use as many consumers as local CPUs
+	// with a max of `maxDefaultStreamBufferConsumers`
+	if numConsumers == 0 {
+		if numConsumers = runtime.NumCPU(); numConsumers > maxDefaultStreamBufferConsumers {
```

Comment:
Would it make sense to use `runtime.GOMAXPROCS(-1)` instead of `runtime.NumCPU`?

Reply:
I think so, although it could be beneficial to have more goroutines than CPUs (after all, goroutines aren't threads).